src/coreclr/src/gc/gc.cpp

```diff
@@ -34117,7 +34120,7 @@ void gc_heap::background_sweep()
 
         if (i == max_generation)
         {
-            start_seg = ephemeral_heap_segment;
+            start_seg = eph_seg;
```

Comment:
this is already saved as `saved_sweep_ephemeral_seg` in `background_ephemeral_sweep`

Reply:
I did not notice that we are already saving that. Will use that one then

Reply:
Thanks!